controllers/managedocs_controller_test.go

```diff
@@ -353,7 +360,138 @@ var _ = Describe("ManagedOCS controller", func() {
+			smtpSecretExists = true
+		} else if errors.IsNotFound(err) {
+			smtpSecretExists = false
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if shouldSMTPSecretExist {
+			if hasValueForSMTP {
+				smtpSecret.Data["host"] = []byte("test-host")
+				smtpSecret.Data["port"] = []byte("20")
+				smtpSecret.Data["username"] = []byte("test-key")
+				smtpSecret.Data["password"] = []byte("test-password")
+			} else {
+				smtpSecret.Data["host"] = []byte("")
+				smtpSecret.Data["port"] = []byte("")
+				smtpSecret.Data["username"] = []byte("")
+				smtpSecret.Data["password"] = []byte("")
+			}
+			if smtpSecretExists {
+				Expect(k8sClient.Update(ctx, smtpSecret)).Should(Succeed())
+			} else {
+				Expect(k8sClient.Create(ctx, smtpSecret)).Should(Succeed())
+			}
+
+			// update notification email to the addon param secret
+			secret := addonParamsSecretTemplate.DeepCopy()
+			secretKey := utils.GetResourceKey(secret)
+			Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+			for i := 0; i < NumOfNotificationEmailAdress; i++ {
+				secret.Data[notificationEmailKeyPrefix+fmt.Sprintf("-%v", i)] = []byte(fmt.Sprintf("test-%v@email.com", i))
+			}
+			Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+		} else if smtpSecretExists {
+			Expect(k8sClient.Delete(ctx, smtpSecret)).Should(Succeed())
+		}
+
+		dmsSecret := dmsSecretTemplate.DeepCopy()
+		var dmsSecretExists bool
+		if err := k8sClient.Get(ctx, utils.GetResourceKey(dmsSecret), dmsSecret); err == nil {
+			dmsSecretExists = true
+		} else if errors.IsNotFound(err) {
+			dmsSecretExists = false
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if shouldPagerdutySecretExist {
+			if hasSnitchUrl {
+				dmsSecret.Data["SNITCH_URL"] = []byte("test-url")
+			} else {
+				dmsSecret.Data["SNITCH_URL"] = []byte("")
+			}
+			if dmsSecretExists {
+				Expect(k8sClient.Update(ctx, dmsSecret)).Should(Succeed())
+			} else {
+				Expect(k8sClient.Create(ctx, dmsSecret)).Should(Succeed())
+			}
+		} else if dmsSecretExists {
+			Expect(k8sClient.Delete(ctx, dmsSecret)).Should(Succeed())
+		}
+
+		pdSecret := pdSecretTemplate.DeepCopy()
+		var pdSecretExists bool
+		if err := k8sClient.Get(ctx, utils.GetResourceKey(pdSecret), pdSecret); err == nil {
+			pdSecretExists = true
+		} else if errors.IsNotFound(err) {
+			pdSecretExists = false
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if shouldPagerdutySecretExist {
+			if hasPagerdutyKey {
+				pdSecret.Data["PAGERDUTY_KEY"] = []byte("test-key")
+			} else {
+				pdSecret.Data["PAGERDUTY_KEY"] = []byte("")
+			}
+			if pdSecretExists {
+				Expect(k8sClient.Update(ctx, pdSecret)).Should(Succeed())
+			} else {
+				Expect(k8sClient.Create(ctx, pdSecret)).Should(Succeed())
+			}
+		} else if pdSecretExists {
+			Expect(k8sClient.Delete(ctx, pdSecret)).Should(Succeed())
+		}
+	}
+
 	Context("reconcile()", func() {
+		When("there is a rook-ceph-operator-config ConfigMap", func() {
```

Comment:
Let's move this test after the addon params test, this way we could assume that the addon params are already in a correct state.

In order to allow the addon params to test creation of resources, we will need to make sure that a `rook-ceph-operator-config` config map exists before the first test